src/firestore.rules.test.js

```diff
@@ -157,10 +163,186 @@ test('allows scholarship delete when user is admin', async () => {
+  });
+
+  test('fails when no scholarship dateAdded', () => {
+    const scholarship = { ...newScholarship };
+    delete scholarship.dateAdded;
+    return assertFails(collection.set(scholarship));
+  });
+
+  test('fails when no scholarship lastModified', () => {
+    const scholarship = { ...newScholarship };
+    delete scholarship.lastModified;
+    return assertFails(collection.set(scholarship));
+  });
+
+  test('fails when scholarship author not an object', () => {
+    return assertFails(collection.set({ ...newScholarship, author: '{}' }));
+  });
+
+  test('fails when author id not a string', () => {
+    return assertFails(
+      collection.set({
+        ...newScholarship,
+        author: { id: 2, email: 'test@gmail.com' },
+      })
+    );
+  });
+
+  test('fails when author email not a string', () => {
+    return assertFails(
+      collection.set({ ...newScholarship, author: { email: 2, id: aliceId } })
+    );
+  });
+
+  describe('requirements validation', () => {
+    const scholarship = { ...newScholarship };
+    scholarship.requirements = {};
+
+    test('fails when scholarship requirements not an object', () => {
+      return assertFails(
+        collection.set({ ...newScholarship, requirements: 3.4 })
+      );
+    });
+
+    test('fails when scholarship gpa is not a number', () => {
+      scholarship.requirements.gpa = '3.0';
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when gpa is less than 0', () => {
+      scholarship.requirements.gpa = -3;
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when gpa is greater than 4', () => {
+      scholarship.requirements.gpa = 4.1;
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when scholarship ethnicities is not an array', () => {
+      scholarship.requirements.ethnicities = '[]';
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when scholarship majors is not an array', () => {
+      scholarship.requirements.majors = 'History';
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when scholarship schools is not an array', () => {
+      scholarship.requirements.schools = 'UCI';
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when scholarship grades is not an array', () => {
+      scholarship.requirements.grades = 8;
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when scholarship states is not an array', () => {
+      scholarship.requirements.states = 'WA';
+      return assertFails(collection.set(scholarship));
+    });
+
+    test('fails when scholarship genders is not an array', () => {
+      scholarship.requirements.genders = 'Male';
+      return assertFails(collection.set(scholarship));
+    });
+  });
+});
+
+describe('validation rules - update', () => {
```

Comment:
These don't seem specific to update? You can group these with the other one. I don't think we need to have two sets of tests for update and create. Just one set and maybe a test case that tests update goes through the same validation failure process